javascript/load-gen.ts

```diff
@@ -0,0 +1,408 @@
+          workerId + 1,
+          numRequestsPerWorker
+        )
+    );
+    const allResultPromises = Promise.all(asyncGetSetResults);
+    await allResultPromises;
+    this.logger.info('DONE!');
+    // wait a few millis to allow the logger to finish flushing
+    await delay(500);
+  }
+
+  private async launchAndRunWorkers(
+    client: SimpleCacheClient,
+    loadGenContext: BasicJavasScriptLoadGenContext,
+    workerId: number,
+    numRequests: number
+  ): Promise<number> {
+    let successCount = 0;
+    for (let i = 1; i <= numRequests; i++) {
+      const result = await this.issueAsyncSetGet(
+        client,
+        loadGenContext,
+        workerId,
+        i
+      );
+      loadGenContext.globalRequestCount++;
+      // TODO: this could be simplified and made more generic, worth doing if we ever want to
+      //  expand this to additional types of behavior
+      switch (result) {
+        case AsyncSetGetResult.SUCCESS:
+          successCount++;
+          loadGenContext.globalSuccessCount++;
+          break;
+        case AsyncSetGetResult.UNAVAILABLE:
+          loadGenContext.globalUnavailableCount++;
+          break;
+        case AsyncSetGetResult.DEADLINE_EXCEEDED:
+          loadGenContext.globalDeadlineExceededCount++;
+          break;
+        case AsyncSetGetResult.RESOURCE_EXHAUSTED:
+          loadGenContext.globalResourceExhaustedCount++;
+          break;
+        case AsyncSetGetResult.RST_STREAM:
+          loadGenContext.globalRstStreamCount++;
+          break;
+      }
+      if (loadGenContext.globalRequestCount % 1000 === 0) {
+        this.logger.info(`
+stats:
+       total: ${
+         loadGenContext.globalRequestCount
+       } (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalRequestCount
+        )} tps)
+     success: ${
+       loadGenContext.globalSuccessCount
+     } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )}%) (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )} tps)
+ unavailable: ${
+   loadGenContext.globalUnavailableCount
+ } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalUnavailableCount
+        )}%)
+deadline exceeded: ${
+          loadGenContext.globalDeadlineExceededCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalDeadlineExceededCount
+        )}%)
+resource exhausted: ${
+          loadGenContext.globalResourceExhaustedCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalResourceExhaustedCount
+        )}%)
+  rst stream: ${
+    loadGenContext.globalRstStreamCount
+  } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalRstStreamCount
+        )}%)
```

Comment:
I think the alignment here is weird?
```
[1657846917254] INFO (load-gen/41589 on griddle.local):
stats:
       total: 9000 (286 tps)
     success: 8922 (99.1%) (283 tps)
 unavailable: 0 (0%)
deadline exceeded: 78 (0.9%)
resource exhausted: 0 (0%)
  rst stream: 0 (0%)

```

Reply:
blech thanks i'll fix this in another PR